arch/common/src/Makefile

```diff
@@ -0,0 +1,27 @@
+############################################################################
+# arch/common/src/Makefile
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.  The
+# ASF licenses this file to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
+# Linker Scripts
+
+ifeq ($(CONFIG_CYGWIN_WINTOOL),y)
+LDFLAGS += $(addprefix -T,$(foreach SCRIPT,$(ARCHSCRIPT),${shell cygpath -w $(SCRIPT)}))
```

Comment:
it's strange that the common folder just have one script file, how about we add a make function in Config.mk to extend the path with cygpath, and invoke that function like this:
```
LDFLAGS += $(addprefix -T $(xxx, $(ARCHSCRIPT)))
```
The new function can be used in many similar place.

Reply:
Yes, how about we put the share script to either:
./tools/
./arch/
We can even invoke arch/Makefile from toplevel Makefile and then:

1. Define the common setting
2. Include the arch specific Makefile

Reply:
Yes, how about put the share script to:
./tools/
./arch/
we can even invoke arch/Makefile and let's arch/Makefile include the arch specific Makefile.

Reply:
That's a good idea. I'll play around with it and see how it looks